lib/ts/hugepages.h

```diff
@@ -20,13 +20,26 @@
  */
 #ifndef _hugepages_h_
 #define _hugepages_h_
+#include "ink_memory.h"
 
 #include <cstring>
 
+#define HUGEPAGE_SIZE  (PAGE_SIZE<<9)
+
 size_t ats_hugepage_size(void);
 bool ats_hugepage_enabled(void);
 void ats_hugepage_init(int);
 void *ats_alloc_hugepage(size_t);
 bool ats_free_hugepage(void *, size_t);
+void *ats_alloc_hugepage_stack(size_t);
+
+void *huge_normal_alloc(void *chunk, size_t size, size_t alignment, bool *zero, bool *commit, unsigned arena_ind);
+void *huge_nodump_alloc(void *chunk, size_t size, size_t alignment, bool *zero, bool *commit, unsigned arena_ind);
+bool huge_dalloc(void *chunk, size_t size, bool committed, unsigned arena_ind);
+bool huge_commit(void *chunk, size_t size, size_t offset, size_t length, unsigned arena_ind);
+bool huge_decommit(void *chunk, size_t size, size_t offset, size_t length, unsigned arena_ind);
+bool huge_purge(void *chunk, size_t size, size_t offset, size_t length, unsigned arena_ind);
+bool huge_split(void *chunk, size_t size, size_t size_a, size_t size_b, bool committed, unsigned arena_ind);
+bool huge_merge(void *chunk_a, size_t size_a, void *chunk_b, size_t size_b, bool committed, unsigned arena_ind);
```

Comment:
And yes ... if we ever demand "this must be a hugepage" ... we are always always going to have to have them assembled for us and will have to wait.

Since we're not in the hardware business, we don't need that kind of latency.